theme/theme.inc

```diff
@@ -60,6 +60,11 @@ function islandora_preprocess_islandora_default_edit(array &$variables) {
           'datastream' => $ds,
         )),
       );
+    }
+    else if (user_access(ISLANDORA_VIEW_DATASTREAM_HISTORY)) {
```

Comment:
So the reason we need an elseif instead of an else is the whole reason this bug existed in the first place. The scenario is that you are viewing the object if you have `ISLANDORA_VIEW_DATASTREAM_HISTORY` but you don't have access to the datastream with `islandora_datastream_access` in that case we were getting a broken table. 

That is because here:
https://github.com/jonathangreen/islandora/blob/aa7566ab10dbe1c64a3d73871d0377b061ddb8a1/theme/theme.inc#L24-L26 we check if you have `ISLANDORA_VIEW_DATASTREAM_HISTORY` access, and add an extra header, but later on we were checking if you have `islandora_datastream_access` and adding the extra row. You can have one but not the other, so we catch that case, and add an extra blank row for it. So what I had before worked, but maybe its more clear with the elseif.